engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/GameManifestProvider.java

```diff
@@ -48,55 +50,91 @@ private GameManifestProvider() {
     /**
      * Generates game manifest with default settings (title, seed) if not specified.
      * Uses default world generator, and modules selection.
-     * @TODO: rewrite/fix it when code will be more stable
      *
-     * @param universeWrapper  contains the universe level properties
-     * @param moduleManager    resolves modules
-     * @param config           provides default module selection, world generator
-     * @return                 game manifest with default settings
+     * @param universeWrapper contains the universe level properties
+     * @param moduleManager   resolves modules
+     * @param config          provides default module selection, world generator
+     * @return game manifest with default settings
+     * @TODO: rewrite/fix it when code will be more stable
      */
-    public static GameManifest createGameManifest(final UniverseWrapper universeWrapper, final ModuleManager moduleManager, final Config config) {
+    public static GameManifest createGameManifest(final UniverseWrapper universeWrapper,
+                                                  final ModuleManager moduleManager,
+                                                  final Config config) {
+
         GameManifest gameManifest = new GameManifest();
-        if (StringUtils.isNotBlank(universeWrapper.getGameName())) {
-            gameManifest.setTitle(universeWrapper.getGameName());
-        } else {
-            gameManifest.setTitle(GameProvider.getNextGameName());
+
+        try {
+            addModulesTo(gameManifest, new DependencyResolver(moduleManager.getRegistry()), config);
+        } catch (DependencyResolutionFailedException e) {
+            logger.error(e.getMessage());
+            return null;
         }
 
+        gameManifest.setTitle(getGameTitle(universeWrapper.getGameName()));
+        gameManifest.setSeed(getSeed(universeWrapper));
+
+        // This is multiplied by the number of seconds in a day (86400000) to determine the exact  millisecond at which the game will start.
+        long startTime = (long) (WorldTime.DAY_LENGTH * WorldTime.NOON_OFFSET);
```

Comment:
I know this PR moved this comment, it's not responsible for how it was written in the first place, but since it's here:

I saw it before and was left confused. Is something dependent on knowing or setting the start time to millisecond precision? what?

Reply:
To be honest: I have no idea :roll_eyes: I hope not...